### PR TITLE
fix: crash when exiting backup dialog [AR-2987]

### PR DESCRIPTION
**Pull request by Garzas**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2987" title="AR-2987" target="_blank">BugAR-2987</a>  Exit restore backup dialog crash
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

When user clicks outside restore from backup dialog app crashes. We are not checking if `latestImportedBackupTempPath` was initialised when closing the dialog.

### Solutions

Check if `latestImportedBackupTempPath`  was initialised. 

#### How to Test

1. Open restore from backup dialog
2. Click on the area outside the dialog

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1387/1/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/settings/backup/BackupAndRestoreViewModel.kt [hunk 1]
```diff
@@ -232,7 +232,11 @@ class BackupAndRestoreViewModel
             backupRestoreProgress = BackupRestoreProgress.InProgress(),
             restorePasswordValidation = PasswordValidation.NotVerified
         )
-        if (kaliumFileSystem.exists(latestImportedBackupTempPath)) kaliumFileSystem.delete(latestImportedBackupTempPath)
+        if (this@BackupAndRestoreViewModel::latestImportedBackupTempPath.isInitialized
+            && kaliumFileSystem.exists(latestImportedBackupTempPath)
+        ) {
+            kaliumFileSystem.delete(latestImportedBackupTempPath)
+        }
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/ui/home/settings/backup/BackupAndRestoreViewModel.kt:239]:
```suggestion
        if (this@BackupAndRestoreViewModel::latestImportedBackupTempPath.isInitialized
                && kaliumFileSystem.exists(latestImportedBackupTempPath)) {
            kaliumFileSystem.delete(latestImportedBackupTempPath)
        }
```

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1387/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1387/3/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/settings/backup/BackupAndRestoreViewModel.kt (hunk 1 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/home/settings/backup/BackupAndRestoreViewModel.kt:239]: In my IDE it's reformated to my changes 😅

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1387/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5571169)(5571169)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv0.0.1(5571169)(5571169)-staging-compat.apk)

File: config/detekt/baseline.xml [hunk 2]
```diff
@@ -20,622 +20,1461 @@
...
+        <ID>LibraryCodeMustSpecifyReturnType:CoreLogicModule.kt$UseCaseModule$@ViewModelScoped @Provides fun provideRestartSlowSyncProcessForRecoveryUseCase( @CurrentAccount currentAccount: UserId, clientScopeProviderFactory: ClientScopeProvider.Factory )</ID>
+        <ID>LibraryCodeMustSpecifyReturnType:CoreLogicModule.kt$UseCaseModule$@ViewModelScoped @Provides fun provideRestoreBackupUseCase(@KaliumCoreLogic coreLogic: CoreLogic, @CurrentAccount currentAccount: UserId)</ID>
+        <ID>LibraryCodeMustSpecifyReturnType:CoreLogicModule.kt$UseCaseModule$@ViewModelScoped @Provides fun provideSelfClientsUseCase(@CurrentAccount currentAccount: UserId, clientScopeProviderFactory: ClientScopeProvider.Factory)</ID>
+        <ID>LibraryCodeMustSpecifyReturnType:CoreLogicModule.kt$UseCaseModule$@ViewModelScoped @Provides fun provideSetUserHandleUseCase(@KaliumCoreLogic coreLogic: CoreLogic, @CurrentAccount currentAccount: UserId)</ID>
+        <ID>LibraryCodeMustSpecifyReturnType:CoreLogicModule.kt$UseCaseModule$@ViewModelScoped @Provides fun provideUpdateApiVersionsScheduler(@KaliumCoreLogic coreLogic: CoreLogic)</ID>
```
Comment (vitorhugods) [inline config/detekt/baseline.xml:156]: This is weird....

@yamilmedina any guesses on why `LibraryCodeMustSpecifyReturnType` is being tracked by Detekt on Reloaded?

**Comment by gongracr** [comment]
These changes are already included in https://github.com/wireapp/wire-android-reloaded/pull/1383 so I'm closing this PR instead.

**Pull request closed without merging**